test/domains/project-knowledge/project-knowledge-store.test.ts

```
@@ -341,12 +465,242 @@ describe('project knowledge local store', () => {
+      await fs.rm(storageRoot, { recursive: true, force: true });
+    }
+  });
+
+  test('rolls back generated history maintenance when legacy payloads are invalid', async () => {
+    const root = await temporaryRoot('comet-project-knowledge-maintenance-rollback-');
+    const storageRoot = await temporaryRoot(
+      'comet-project-knowledge-maintenance-rollback-storage-',
+    );
+    const diagnostics: Array<{ code: string; message: string }> = [];
+    let store: ProjectKnowledgeLocalStore | undefined;
+    let databasePath = '';
```

ESLint

[error] The value assigned to 'databasePath' is not used in subsequent statements.

(no-useless-assignment)